app/models/practice.rb

```diff
@@ -181,4 +217,20 @@ def number_of_adopted_facilities
   def date_range_views(start_date, end_date)
     Ahoy::Event.where_props(practice_id: id).where(time: start_date..end_date).count
   end
+
+  def favorited_count
+    user_practices.where({favorited: true}).count
+  end
+
+  def favorited_count_by_range(start_date, end_date)
+    user_practices.where({created_at: start_date..end_date, favorited: true}).count
```

Comment:
this is a note for myself: we may have to rethink how the favorites work since we want to track by time.
the `created_at` for a `UserPractice` could be created when the user "commits to" a practice, and then they can favorite it later, and vice versa. 
we may need something like `date_favorited` and `date_committed`.